common/metrics/metrics.go

```diff
@@ -26,17 +26,19 @@ package metrics
 
 import (
 	"time"
+
+	"go.temporal.io/server/common/log"
 )
 
 // Mostly cribbed from
 // https://github.com/temporalio/sdk-go/blob/master/internal/common/metrics/handler.go
 // and adapted to depend on golang.org/x/exp/event
 type (
-	// MetricProvider represents the main dependency for instrumentation
-	MetricProvider interface {
+	// MetricHandler represents the main dependency for instrumentation
+	MetricHandler interface {
```

Comment:
Since we try to match with SDK, let do exact match as MetricsHandler: https://github.com/temporalio/sdk-go/blob/706af9738b9c79acc58212364f3c864935c90b78/client/client.go#L433